### PR TITLE
    Fix image_resize nodelet

### PR DESCRIPTION
**Pull request by bikramak**
* Update interpolation types
* Add arguments to enable disable each nodelet
* Add default arguments for image_resize and image_rect
* Use toCVShare instead of toCVCopy
* Include image_resize in image_proc

### CODE REVIEW

File: image_proc/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,16 @@
 cmake_minimum_required(VERSION 2.8)
 project(image_proc)
 
-find_package(catkin REQUIRED)
+find_package(catkin REQUIRED
```
Comment (jacobperron) [inline image_proc/CMakeLists.txt]: `REQUIRED COMPONENTS`

File: image_proc/cfg/Resize.cfg [hunk 2]
```diff
@@ -8,30 +8,30 @@ ID = 'Resize'
...
-                             gen.const('Lanczos4', int_t, 4, 'Lanczos4')],
-                            'interpolation type')
-gen.add('interpolation', int_t, level=0,
-        description='Interpolation algorithm between source image pixels',
+interpolate_enum = gen.enum([gen.const("INTER_NEAREST", int_t, 0, "Nearest neighbor"),
+                             gen.const("INTER_LINEAR", int_t, 1, "Linear"),
+                             gen.const("INTER_CUBIC", int_t, 2, "Cubic"),
+                             gen.const("INTER_LANCZOS4", int_t, 4, "Lanczos4")],
+                            "interpolation type")
+gen.add("interpolation", int_t, level=0,
```
Comment (jacobperron) [inline image_proc/cfg/Resize.cfg:16]: All of these single to double quote conversions seem unnecessary.

File: image_proc/launch/image_proc.launch [hunk 3]
```diff
@@ -1,29 +1,64 @@
...
+  </node>
+
+  <!-- Color image resize-->
+  <node  if="$(arg image_resize)" pkg="nodelet" type="nodelet" name="resize_image"
+        args="load image_proc/resize $(arg manager) $(arg bond)"
+        respawn="$(arg respawn)">
+        <param name="scale_height" value="$(arg image_resize_factor)" />
+        <param name="scale_width" value="$(arg image_resize_factor)" />
+        <remap from="in_image" to="$(arg resize_in_topic)" />
+        <remap from="resize_image/out_image" to="$(arg resize_out_topic)" />
```
Comment (jacobperron) [inline image_proc/launch/image_proc.launch]: Why is the out topic namepaced and not the input topic? I guess it should be fine to remove the namespace.

File: image_proc/launch/image_proc.launch [hunk 4]
```diff
@@ -1,29 +1,64 @@
...
-  </node>  
+    <remap from="image_mono" to="$(arg color_rect_in_topic)" />
+    <remap from="image_rect" to="$(arg color_rect_out_topic)" />
+  </node>
+
+  <!-- Color image resize-->
+  <node  if="$(arg image_resize)" pkg="nodelet" type="nodelet" name="resize_image"
+        args="load image_proc/resize $(arg manager) $(arg bond)"
+        respawn="$(arg respawn)">
+        <param name="scale_height" value="$(arg image_resize_factor)" />
```
Comment (jacobperron) [inline image_proc/launch/image_proc.launch]: Over-indentation.

File: image_proc/CMakeLists.txt [hunk 5]
```diff
@@ -12,11 +19,16 @@ if(cv_bridge_VERSION VERSION_GREATER "1.12.0")
...
+  cfg/CropDecimate.cfg
+  cfg/Debayer.cfg
+  cfg/Rectify.cfg
+  cfg/Resize.cfg)
 
-catkin_package(
-  CATKIN_DEPENDS image_geometry roscpp sensor_msgs
-  DEPENDS OpenCV
+catkin_package(CATKIN_DEPENDS
+  image_geometry
```
Comment (jacobperron) [inline image_proc/CMakeLists.txt]: Although all of this CMake reformatting is probably best left to the maintainer, I would have formatted as such:

```
catkin_package(
  INCLUDE_DIRS
    include
  LIBRARIES
    ${PROJECT_NAME}
  CATKIN_DEPENDS
    image_geometry
    roscpp
    sensor_msgs
)
```

:stuck_out_tongue:

**Review by jacobperron — APPROVED** [review]
The core of the change looks good to me. Some minor comments.

File: image_proc/launch/image_proc.launch [hunk 6]
```diff
@@ -1,29 +1,64 @@
...
+  <!-- Manager name must be globally qualified -->
+  <arg name="manager" /> 
+  <!-- Worker threads -->
+  <arg name="num_worker_threads" default="4"/>
 
-  <arg name="manager" /> <!-- Must be globally qualified -->
-  <arg name="respawn" default="false" />
-  <!-- TODO Arguments for debayer, interpolation methods? -->
+  <arg name="respawn"               default="false" />
+  <arg name="debayer"               default="true" />
```
Comment (SomeshDaga) [inline image_proc/launch/image_proc.launch:10]: What does `debayer` do and do we want this `true` by default? Same with `monochrome_rect` and `color_rect` below

**Review by SomeshDaga — APPROVED** [review]

File: image_proc/CMakeLists.txt (hunk 5 above)
Comment (bikramak) [reply image_proc/CMakeLists.txt]: Thanks for the suggestion. I have incorporated it ;).
I will push maintainer to follow this style. I am open to suggestion and review from maintainer, if they consider it inappropriate to merge it with this style.

File: image_proc/cfg/Resize.cfg (hunk 2 above)
Comment (bikramak) [reply image_proc/cfg/Resize.cfg:16]: This single quotes is not consistent with other nodelet config files. I am trying help keep code consistent.

File: image_proc/launch/image_proc.launch (hunk 4 above)
Comment (bikramak) [reply image_proc/launch/image_proc.launch]: Fixed.

File: image_proc/launch/image_proc.launch (hunk 3 above)
Comment (bikramak) [reply image_proc/launch/image_proc.launch]: Nope. The out_image publisher is namespaced within the nodelet itself wheres the in_image is not. I have tried it without namespace and that did not work.

File: image_proc/launch/image_proc.launch (hunk 6 above)
Comment (bikramak) [reply image_proc/launch/image_proc.launch:10]: I have left the parameters values such that it is set to whatever it was there previously without using image_resize. I hope it helps keep the package backwards compatible.

**Pull request merged**